### PR TITLE
Added file to fetch tweets favorited by user

### PR DESCRIPTION
**Pull request by voldyman**
just need to add the twitter keys to a config file and run fav-tweets.php as a cron job.

we will get a json file tweets.json with all the required info.

How-to:
* Create a Twitter app from http://dev.twitter.com
* Edit `backend/config.php`, write consumer key & secret
* See https://dev.twitter.com/oauth/overview/application-owner-access-tokens to get access token & secret

### CODE REVIEW

**Comment by gamerlv** [comment]
Instead of requiring a new config file could you please use the existing `backend/config.php`

It looks like this file is never to be called from the user end, if that is so some method of restricting it to a few IP's (eg 127.0.0.1) or just CLI would be usefull

**Comment by emersion** [comment]
How-to:
* Create a Twitter app from http://dev.twitter.com
* Edit `backenbd/config.php`, write consumer key & secret
* See https://dev.twitter.com/oauth/overview/application-owner-access-tokens to get access token & secret

- [ ] PHP notice when running from CLI: `PHP Notice:  Undefined index: REMOTE_ADDR in /home/simon/public_html/mvp/backend/fav-tweets.php on line 10`
- [ ] Could not test from localhost - `$_SERVER['REMOTE_ADDR']` was `::1`
- [ ] 4 spaces indentation - see https://github.com/elementary/mvp#code-style
- [ ] I executed the script from the root (`php -f backend/fav-tweets.json`), so the `tweets.json` file was written to the root instead of `backend/`
- [ ] Try not to mix `'` and `"` - use only `'` and `'string'.$var`
- [ ] After executing the script, `tweets.json` only contained `[]` - maybe because I'm not logged in with the elementary account?
- [ ] It's generally a good practice to use protected/private variables in classes. Use `protected` instead of `var`.

Thanks! :-)

**Comment by voldyman** [comment]
For the 6th point, you need to add keys in the config file

**Comment by emersion** [comment]
I've added keys in the config file. Do you execute the script with your own Twitter account?

**Comment by voldyman** [comment]
yes, i have tested that

File: backend/fav-tweets.php [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        $this->consumer_secret = $config['twitter_consumer_secret'];
+
+        $this->oauth = array(
+            'oauth_consumer_key' => $this->consumer_key,
+            'oauth_token' => $this->access_token,
+            'oauth_signature_method' => 'HMAC-SHA1',
+            'oauth_timestamp' => time(),
+            'oauth_nonce' => time(),
+            'oauth_version' => '1.0'
+            );
```
Comment (emersion) [inline backend/fav-tweets.php]: Indentation problem here ;-)

File: backend/fav-tweets.php [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        $request = 'users/lookup.json';
+        $params = array('user_id'=>$list);
+
+        $result = $this->send_request($request, $params);
+        return $result;
+    }
+
+    function get_favorites($user_id, $count = 10) {
+        $request = 'favorites/list.json';
+        $params = array('user_id' => $user_id,
```
Comment (emersion) [inline backend/fav-tweets.php]: Change this to `screen_name`.

See https://dev.twitter.com/rest/reference/get/favorites/list

**Comment by emersion** [comment]
LGTM, thanks a lot! :-D

**Pull request merged**